pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -190,7 +191,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(
 	util.SortVirtualHosts(virtualHosts)
 
 	if !useSniffing {
-		virtualHosts = append(virtualHosts, node.CatchAllVirtualHost)
+		// virtualhost envoyfilter can mutate this sharing config.
+		catchAll := protobuf.Clone(node.CatchAllVirtualHost).(*route.VirtualHost)
```

Comment:
the whole point of `node.CatchAllVirtualHost` was to cache it. So if we are not caching it, why not just revert https://github.com/istio/istio/commit/b32710c73525c0ba0df8af1d4903d947da68e178?

My belief (untested!) is that `proto.Clone` is slower than rebuilding from scratch

Reply:
Good question, I feel clone may be quicker than rebuild. let me test it

Reply:
You are totally right. clone is more slower

BenchmarkClone-8                	  549873	      2212 ns/op
BenchmarkBuild-8                	 1844498	       643.6 ns/op



```
+
+
+func BenchmarkClone(b *testing.B) {
+       vhost := istionetworking.BuildCatchAllVirtualHost(true, "")
+       for i := 0; i < b.N; i++ {
+               _ = protobuf.Clone(vhost).(*route.VirtualHost)
+       }
+}
+
+func BenchmarkBuild(b *testing.B) {
+       for i := 0; i < b.N; i++ {
+               _ = istionetworking.BuildCatchAllVirtualHost(true, "")
+       }
+}

```